codingStandards/CodingStandard-CSharp.md

```diff
@@ -0,0 +1,367 @@
+Hungarian notation is a defined set of pre and postfixes which are applied to names to reflect the type of the variable. This style was used in early Windows programming, but is now obsolete.
+
+![](Good.png)
+
+```cs
+Name
+Colors
+```
+
+![](Bad.png)
+
+```cs
+strName
+ColorsEnum
+```
+
+> **Exception**: All fields and variable names that contain GUI elements like button should be postfixed with their type name without abbreviations. e.g., `cancelButton`, `nameTextBox`.
+
+
+## Commenting Conventions
+####1. General guidelines.
+
+- Place the comment on a separate line, not at the end of a line of code.
+- Begin comment text with an upper case letter.
+- Insert one space between comment delimiter (`//`) and comment text.
+- Use `//` or `///` but never `/* ... */`.
+- The length of comment should not exceed the length of code.
+
+####2. Document all public, protected and internal types and members.
+Documenting your code allows Visual Studio to pop-up the documentation when your class is used somewhere else. You can form your documentation using [XML tags](https://msdn.microsoft.com/en-us/library/5ast78ax.aspx).
+
+```cs
+/// <summary>
+/// Get a value indicating whether the user has a license.
+/// </summary>
+/// <returns>
+/// <c>true</c> if the user has a license; otherwise <c>false</c>.
+/// </returns>
+public bool HasLicense() { ... }
+```
+
+## Language Conventions
+####1. Do not omit access modifiers.
+Explicitly declare all identifiers with the appropriate access modifiers instead of allowing the default.
+
+![](Good.png)
+
+```cs
+private void WriteEvent(string message)
+```
+
+![](Bad.png)
+
+```cs
+void WriteEvent(string message)
+```
+
+####2. Always use the built-in C# data type aliases, instead of the .NET common type system.
+
+![](Good.png)
+
+```cs
+short
+int
+long
+string
+```
+
+![](Bad.png)
+
+```cs
+Int16
+Int32
+Int64
+String
+```
+
+####3. Only use var when the type is very obvious.
+When the type of a variable is clear from the context, use var in the declaration.
+
+```cs
+var welcomeMessage = "This is a welcome message!";
+var account = new Account();
+```
+
+Do not use var when the type is not apparent from the right side of the assignment.
+
+```cs
+int result = ExampleClass.ResultSoFar();
+```
+> To know more about when to use/not to use implicit typing read [Uses and misuses of implicit typing](https://blogs.msdn.microsoft.com/ericlippert/2011/04/20/uses-and-misuses-of-implicit-typing/).
+
+####4. Favor Object and Collection initializers over separate statements.
+
+![](Good.png)
+
+```cs
+var startInfo = new ProcessStartInfo("myapp.exe");
+{
+	StandardOutput = Console.Output,
+	UseShellExecute = true
+};
+```
+
+![](Bad.png)
+
+```cs
+var startInfo = new ProcessStartInfo("myapp.exe");
+startInfo.StandardOutput = Console.Output;
+startInfo.UseShellExecute = true;
+```
+
+## References
+1. [C# Coding Coventions (C# Programming Guide)](https://msdn.microsoft.com/en-us/library/ff926074.aspx)<br>
```

Comment:
Perhaps we can give some info about each link? e.g.

1. [C# Coding Coventions (C# Programming Guide)](https://msdn.microsoft.com/en-us/library/ff926074.aspx) -- From Microsoft<br>